### DISCUSSION TITLE
Opening Malformed TSV Files

### DISCUSSION THREAD

**Discussion by andrewblount** [dathere/qsv#1070] [Q&A] [answered]
Hello!  I love your qsv application. Thank you for your hard work.  I also love your name. If you look me up on YouTube, you'll understand why. :-)

I'd use it even more often if I could get past a persistent error.  All of my files are TSV with no quotes.  80% plus of the time I try to perform any operation on them, I get an error that one of the rows does not have the same number of columns as the other row.  This is correct.  I deal with tens of thousands of files, many are malformed.  However, I'd like to be able to run the operation on the file still and simply pad the rows that are missing columns.  I've read through everything online and can't find a solution other than to use a different application to standardize the number of columns before running anything with qsv.  What am I missing?

Thank you in advance for your help.

Andrew

**Comment by ondohotola** [comment, marked as answer]
Have you tried `qsv fixlengths`?

**Comment by andrewblount** [comment]
Thank you for the really fast reply.  I have.  My understanding is that normalized the character lengths, but not the number of columns.  Am I incorrect?

**Comment by ondohotola** [comment]
Try it out on a malformed file

**Comment by andrewblount** [comment]
Thank you!!!  That worked.  I misread the description of it.  You're really kind to help.

**Reply by ondohotola** [reply]
You are most welcome.

**Comment by andrewblount** [comment]
Here's a screenshot in-case anyone else can benefit from seeing it.  

<img width="910" alt="2023-06-24  Screenshot of  iTerm2 andrewblount@hazardouswaste~LocalTempqsv at 06-20 PM" src="https://github.com/jqnatividad/qsv/assets/8727033/15b554bb-e2c2-4810-9b2e-a735d5e42584">
